### PR TITLE
Fix new issues with boundary relationship modeling

### PR DESCRIPTION
**Pull request by vcschapp**
# Description

## Primary change

The primary change in this PR is to align the foreign key references to division features within boundary features with the schema relationships guidance.

In a recent change, PR #200, the schema for the divisions theme's boundary feature fell out of alignment with the *Relationships in the Schema* guidance.
    
The guidance is that a property or nested sub-property that contains a foreign key reference to another feature by Overture ID must end in the suffix `_id`.
    
Before PR #200, things were fine because the Overture ID was contained in a nested sub-property, `divisions[*].division_id`. But since PR #200 removed the nested structures from the `divisions` property, it should have renamed it `division_ids`. This change fixes that.

## Secondary change

In another recent change, PR #188, the boundary feature's geometry property description got decoupled from the actual allowed content of the geometry property. This change fixes the description.

# Reference

- Foreign key relationships:
    - https://github.com/OvertureMaps/schema/pull/200
    - https://github.com/OvertureMaps/schema-wg/blob/admin/decisions/adr-relationship-modeling.md
    - https://wiki.overturemaps.org/x/oAEjAQ
- Description change: https://github.com/OvertureMaps/schema/pull/188

# Testing

Updated examples and counterexamples and ran the test script.

# Checklist

*Checklist of tasks commonly-associated with schema pull requests. Please review the relevant checklists and ensure you do all the tasks that are required for the change you made.*

1. [X] ~~Add~~ Update relevant examples.
2. [ ] Add relevant counterexamples.
3. [X] Update any counterexamples that became obsolete. For example, if a counterexample uses property `A` but is not intended to test property `A`'s validity, and you made a schema change that invalidates property `A` in that counterexample, fix the counterexample to align it with your schema change.  
4. [X] Update in-schema documentation using plain English written in complete sentences, if an update is required.
5. [ ] Update Docusaurus documentation, if an update is required.
6. [ ] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

# Documentation Website

*Update the hyperlink below to put the pull request number in.*

[Docs preview for this PR.](https://dfhx9f55j8eg5.cloudfront.net/pr/215)

### CODE REVIEW

File: schema/divisions/boundary.yaml [hunk 1]
```diff
@@ -30,16 +30,16 @@ properties:     # JSON Schema: Top-level object properties.
...
+          Identifies the two divisions to the left and right, respectively, of
+          the boundary line. The left- and right-hand sides of the boundary are
+          considered from the perspective of a person standing on the line
+          facing in the direction in which the geometry is oriented, i.e. facing
+          toward the end of the line.
           
-          The first array element represents the left division. The second
-          element represents the right division.
+          The first array element is the Overture ID of the left division. The
+          second element is the Overture ID of the right division.
```
Comment (sasastanojkov) [inline schema/divisions/boundary.yaml:42]: Overture ID or GERS id? Since technically division GERS ids will go there.

Comment (vcschapp) [reply schema/divisions/boundary.yaml:42]: My understanding is we're aiming to rename "GERS ID" to "Overture ID" to reduce cognitive load. Same concept, same stable ID, different name.

@jenningsanderson @danabauer Do I have this right?

**Review by jwass — APPROVED** [review]

File: schema/divisions/boundary.yaml (hunk 1 above)
Comment (danabauer) [reply schema/divisions/boundary.yaml:42]: We're in limbo until the Steering Committee weighs in. One possible (and likely) outcome is that we'll refer to the `id` as "Overture ID" and the reference system as GERS.

Comment (danabauer) [reply schema/divisions/boundary.yaml:42]: I agree that you should use "Overture ID"

File: counterexamples/divisions/boundary/bad-division_ids-1.yaml [hunk 2]
```diff
@@ -7,11 +7,11 @@ geometry:
 properties:
   theme: divisions
   type: boundary
-  update_time: "2024-02-23T22:47:06Z"
-  version: 0
+  update_time: "2024-06-12T17:52:42Z"
+  version: 1
```
Comment (danabauer) [inline counterexamples/divisions/boundary/bad-division_ids-1.yaml:11]: this is great to see the version updated here. Do you know if we're doing this consistently across themes?

**Review by danabauer — APPROVED** [review]

**Review by TristanDiet-TomTom — APPROVED** [review]

**Review by DavidKarlas — APPROVED** [review]

File: counterexamples/divisions/boundary/bad-division_ids-1.yaml (hunk 2 above)
Comment (vcschapp) [reply counterexamples/divisions/boundary/bad-division_ids-1.yaml:11]: It's really just me who does this. I can be meticulous to the point of silliness sometimes.

**Pull request merged**